private/buf/buffetch/buffetch.go

```diff
@@ -216,6 +225,12 @@ type ReadWriteBucketCloser internal.ReadWriteBucketCloser
 // ReadBucketCloserWithTerminateFileProvider is a ReadBucketCloser with a TerminateFileProvider.
 type ReadBucketCloserWithTerminateFileProvider internal.ReadBucketCloserWithTerminateFileProvider
 
+// TerminateFileProvider provides TerminateFiles.
+type TerminateFileProvider = internal.TerminateFileProvider
+
+// TerminateFile is a terminate file.
+type TerminateFile = internal.TerminateFile
```

Comment:
These are exported so we can have a mock `buffetch.Reader`

Reply:
Re-unexport these - having to export these to write tests in another package sets off alarm bells for me. It likely means we should pursue a different testing strategy here.

Reply:
unexported in [1c288c4](https://github.com/bufbuild/buf/pull/2490/commits/1c288c4f2e089a52abbe8187cb5c1119b235e5a9)